spec/Appendix C -- Transitional Non-Null.md

```diff
@@ -0,0 +1,129 @@
+# C. Appendix: Transitional Non-Null
+
+Note: This appendix defines an optional mechanism enabling existing fields to be
+marked as `Non-Null` for clients that opt out of error propagation without
+changing the error propagation boundaries for deployed legacy clients.
+Implementations are not required to support this feature, but doing so enables
+gradual migration toward semantic nullability while preserving compatibility.
+
+## Overview
+
+With the introduction of _error behavior_, clients can take responsibility for
+handling of _execution error_: correlating {"errors"} in the result with `null`
+values inside {"data"} and thereby removing the ambiguity that error propagation
+originally set out to solve. If all clients adopt this approach then schema
+designers can, and should, reflect true nullability in the schema, marking
+fields as `Non-Null` based on their data semantics without regard to whether or
+not they might error.
+
+However, legacy clients may not perform this correlation. Introducing `Non-Null`
+in such cases could cause errors to propagate further, potentially turning a
+previously handled error in a single field into a full-screen error in the
+application.
+
+To support a smooth transition, this appendix introduces the `@noPropagate`
+directive and the concept of _transitional_ Non-Null types. These wrappers raise
+errors like regular `Non-Null` types, but suppress propagation and appear
+nullable in introspection when using the legacy {"PROPAGATE"} _error behavior_.
+
+## The @noPropagate Directive
+
+```graphql
+directive @noPropagate(levels: [Int!]! = [0]) on FIELD_DEFINITION
```

Comment:
Some alternative names:

- @semantic - This field type is semantically non-null. It will never have a null value except in case of error, but we since errors happen it may have a null value
- @soft - This is a "soft" non-null in that it is not enforced by propagation

Reply:
* `@transitionalErrorBoundary`
* `@transitionallyNullable`
* `@localError`

??